packages/flutter_tools/bin/macos_assemble.sh

```diff
@@ -93,3 +93,12 @@ RunCommand "${FLUTTER_ROOT}/bin/flutter"                                    \
     --build-outputs="${build_outputs_path}"                                 \
     --output="${ephemeral_dir}"                                             \
    "${build_mode}_macos_bundle_flutter_assets"
+
+if [[ $? -ne 0 ]]; then
+  EchoError "Failed to build ${project_path}."
+  exit -1
```

Comment:
This makes sense for now.
A better solution for both platforms would be to craft one-line error messages (not like the `=======================` errors we have now) and echo prefixed with "error " or "warning ".  This will show up as a real Xcode error instead of just a failure of a script exiting with nonzero, as discussed in chat one time.

![unnamed](https://user-images.githubusercontent.com/682784/90298236-41b9e400-de46-11ea-8324-aa6d0b36934d.png)
![unnamed-1](https://user-images.githubusercontent.com/682784/90298237-42527a80-de46-11ea-8375-8e8afa8bfbbb.png)
except that's supposed to say "This is an error message" and I'm too lazy to recreate this.

Reply:
Is this something we need to do for our `printError` calls? That is not that most difficult thing to add...

Reply:
We are way too chatty with out `printError` calls to just prepend `error `.  For example, I believe every dart compilation failure line is a separate `printError`, including the `  _^_ ` whatever ASCII annotation line that points to the column of the printed line, so there would be like 5 Xcode errors for every dart compilation error, when there should just be one.  I may be misremembering though.

I have a branch for this I never finished, for iOS:
logger.dart
```dart
class XcodeScriptPhaseStdoutLogger extends StdoutLogger {
  XcodeScriptPhaseStdoutLogger({
    @required AnsiTerminal terminal,
    @required Stdio stdio,
    @required TimeoutConfiguration timeoutConfiguration,
    StopwatchFactory stopwatchFactory = const StopwatchFactory(),
  }) : super(
    terminal: terminal,
    stdio: stdio,
    outputPreferences: OutputPreferences(
      wrapText: false,
      showColor: false,
    ),
    timeoutConfiguration: timeoutConfiguration,
    stopwatchFactory: stopwatchFactory,
  );

  @override
  void printError(
      String message, {
        StackTrace stackTrace,
        bool emphasis,
        TerminalColor color,
        int indent,
        int hangingIndent,
        bool wrap,
      }) {
    String xcodeMessage = message;
    if (message.contains('Error:')) { // <-- "Error:" was the best indicator I had for a dart issue.
      xcodeMessage = 'error: $message'; 
    } // <-- Probably need more cases here.

    super.printError(
      xcodeMessage,
      stackTrace: stackTrace,
      emphasis: false,
      color: color,
      indent: indent,
      hangingIndent:
      hangingIndent,
      wrap: wrap
    );
  }
}
```
xcode_backend.sh
```sh
  RunCompileCommand "${FLUTTER_ROOT}/bin/flutter"                                \
    ${verbose_flag}                                                       \
    ${flutter_engine_flag}                                                \
    ${local_engine_flag}                                                  \
    assemble                                                              \
    --output="${derived_dir}/"                                            \
    ${performance_measurement_option}                                     \
    --logging-style=xcode-build-phase                                     \
...
```
assemble.dart
```dart
    argParser.addOption(
      'logging-style',
      hide: true,
      allowed: <String>['xcode-build-phase'],
      help: 'Format logging.',
    );
```

On the right track?

Reply:
Yeah that seems like a good place to start, though I would leave out the message introspection and instead configure it via the calls to printError